packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -834,6 +834,11 @@ class _RenderDecoration extends RenderBox {
   // This method applies layout to all of the renderers except the container.
   // For convenience, the container is laid out in performLayout().
   _RenderDecorationLayout _layout(BoxConstraints layoutConstraints) {
+    assert(
+      layoutConstraints.maxWidth < double.infinity,
+      'A text input cannot have an unbounded width. If you want it to take up the leftover space inside its parent, try wrapping it in a Flexible.',
```

Comment:
Maybe make the report a little more informative:

'An InputDecorator, which is typically created by a TextField, cannot have an unbounded width.\n'
'This happens when the parent widget does not provide a finite width constraint. '
'For example if the InputDecorator is contained by a Row, then its width must be '
'constrained. An Expanded widget or a SizedBox can be used to constrain the width '
'of the InputDecorator or the TextField that contains it.'